pkg/auth/handler/webapp/authflow_controller.go

```diff
@@ -691,6 +691,9 @@ func (c *AuthflowController) takeBranchRecursively(s *webapp.Session, screen *we
 		// This taken branch require an input to select.
 		case webapp.TakeBranchResultInput:
 			output, screen, err = c.handleTakeBranchResultInput(s, screen, takeBranchResult)
+			if err != nil {
+				return
+			}
```

Comment:
What bug does this change fix?

Reply:
The bug here is even taking branch has error, the webapp will still continue the loop and finally panic with nil pointer exception.
So I fixed it and ensure an error will be returned.

However, the real error is actually something more difficult to fix, which I opened an issue to fix later: https://github.com/authgear/authgear-server/issues/3549